cmd/src/actions_exec_logger.go

```diff
@@ -227,9 +225,106 @@ func (a *actionLogger) DockerStepStarted(repoName string, step int, image string
 }
 
 func (a *actionLogger) DockerStepErrored(repoName string, step int, err error, elapsed time.Duration) {
+	a.progress.IncStepsComplete(1)
 	a.write(repoName, boldRed, "%s %s. (%s)\n", boldBlack.Sprintf("[Step %d]", step), err, elapsed)
 }
 
 func (a *actionLogger) DockerStepDone(repoName string, step int, elapsed time.Duration) {
+	a.progress.IncStepsComplete(1)
 	a.write(repoName, yellow, "%s Done. (%s)\n", boldBlack.Sprintf("[Step %d]", step), elapsed)
 }
+
+func (a *actionLogger) write(repoName string, c *color.Color, format string, args ...interface{}) {
+	if w, ok := a.RepoWriter(repoName); ok {
+		fmt.Fprintf(w, format, args...)
+	}
+
+	if a.verbose {
+		if len(repoName) > 0 {
+			format = fmt.Sprintf("%s -> %s", c.Sprint(repoName), format)
+		}
+		fmt.Fprintf(a.progress, format, args...)
+	}
+}
+
+type progress struct {
+	jobs          int64
+	stepsComplete int64
+	totalSteps    int64
+
+	mu          sync.Mutex
+	w           io.Writer
+	shouldClear bool
+}
+
+func (p *progress) Write(data []byte) (int, error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	length := 50
```

Comment:
I'd suggest moving this definition down to where it's first used and calling it `maxLength` or something like that, because `length` without any context could be anything 😄